### PR TITLE
fix: add extract PRETTY_NAME from /etc/os-release to as the current OS

### LINKED ISSUE #6967: SUSE Linux Entreprise Server seen as an Unknown Linux Distribution.
### Summary 
SUSE Linux Entreprise Server seen as an Unknown Linux Distribution.

### Steps to reproduce the behavior
Running  ```jx version```on a SUSE Linux Entreprise Server host

### Expected behavior
Using /etc/os-release instead of uname would provide more infos about the OS : 
```cat /etc/os-release
NAME="SLES"
VERSION="15-SP1"
VERSION_ID="15.1"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP1"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp1"
```

### Jx version
``` jx version
NAME               VERSION
jx                 2.0.1249
Kubernetes cluster v1.15.2
kubectl            v1.15.2
helm client        2.14.1
git                2.16.4
Operating System   Unknown Linux distribution Linux version 4.12.14-lp151.27-default (geeko@buildhost) (gcc version 7.4.0 (SUSE Linux) ) #1 SMP Fri May 10 14:13:15 UTC 2019 (862c838)
```

The output of `jx version` is:

```
same as above
```

### Jenkins type

- [ X] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
refer to #6967 

#### Which issue this PR fixes

fixes #6967

### CODE REVIEW

**Comment by nodece** [comment]
debian: echo `Debian GNU/Linux 10 (buster) `.

File: pkg/util/system/version_linux.go [hunk 1]
```diff
@@ -40,21 +26,43 @@ func GetOsVersion() (string, error) {
...
 }
 
-func runCommand(command string, args ...string) (string, error) {
-	e := exec.Command(command, args...)
-	data, err := e.CombinedOutput()
-	text := string(data)
-	text = strings.TrimSpace(text)
+func extractPrettyNameFromOsRelease() (string, error) {
+	file := "/etc/os-release"
+	buf, err := ioutil.ReadFile(file)
```
Comment (hferentschik) [inline pkg/util/system/version_linux.go]: If we were to call `getFileContents` here as well and we would change `getFileContents` to be an implementation of a say `ReleaseFileGetter` interface, we actually would get into a position where we could test this. 

The default implementation would still read from the file system, but for testing we can return arbitrary data. 

WDYT?

Comment (hferentschik) [reply pkg/util/system/version_linux.go]: Also, one could use the string returned by `getFileContent` and split it at '\n' and pass the slice then to `util.ExtractKeyValuePairs` which returns a map of key/value pairs. Then one just can check for the "PRETTY_NAME" key.

File: pkg/util/system/version_linux.go [hunk 2]
```diff
@@ -40,21 +26,43 @@ func GetOsVersion() (string, error) {
...
+		}
 	}
-	return text, err
+
+	return "", fmt.Errorf("Could not extract PRETTY_NAME from %s", file)
 }
 
 func getFileContents(file string) (string, error) {
-	bytes, err := ioutil.ReadFile(file)
+	buf, err := ioutil.ReadFile(file)
```
Comment (hferentschik) [inline pkg/util/system/version_linux.go]: 👍

File: pkg/util/system/version_linux.go (hunk 1 above)
Comment (nodece) [reply pkg/util/system/version_linux.go]: @hferentschik Yeah, done.

**Comment by nodece** [comment]
I cannot access https://dashboard-jx.jenkins-x.live/teams/jx/projects/jenkins-x/jx/PR-6974/6.

**Comment by hferentschik** [comment]
> I cannot access https://dashboard-jx.jenkins-x.live/teams/jx/projects/jenkins-x/jx/PR-6974/6.

I know :-(

```
--- FAIL: TestGetOsVersionReturnsNoErrorWithMock (0.00s)
    version_linux_test.go:47: 
        	Error Trace:	version_linux_test.go:47
        	Error:      	Not equal: 
        	            	expected: "Ubuntu 18.04.4 LTS"
        	            	actual  : "Unknown Linux distribution Ubuntu 18.04.4 LTS"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-Ubuntu 18.04.4 LTS
        	            	+Unknown Linux distribution Ubuntu 18.04.4 LTS
        	Test:       	TestGetOsVersionReturnsNoErrorWithMock
FAIL
```

**Comment by nodece** [comment]
@hferentschik Thanks for your patient guidance. I've been fix it.

**Comment by hferentschik** [comment]
thanks for the pull request @nodece. Much appreciated.

**Comment by nodece** [comment]
:) I looking forward to more contribution in GSoC 2020.

**Pull request merged**